tests/app/delivery/test_send_to_providers.py

```diff
@@ -271,29 +272,27 @@ def test_should_call_send_sms_response_task_if_research_mode(
     send_to_providers.send_sms_to_provider(sample_notification)
     assert not aws_sns_client.send_sms.called
 
-    app.delivery.send_to_providers.send_sms_response.assert_called_once_with(
-        "sns", str(sample_notification.id), sample_notification.to
-    )
+    app.delivery.send_to_providers.send_sms_response.assert_called_once_with("sns", sample_notification.to)
 
     persisted_notification = notifications_dao.get_notification_by_id(sample_notification.id)
     assert persisted_notification.to == sample_notification.to
     assert persisted_notification.template_id == sample_notification.template_id
     assert persisted_notification.status == "sent"
     assert persisted_notification.sent_at <= datetime.utcnow()
     assert persisted_notification.sent_by == "sns"
-    assert persisted_notification.reference is None
+    assert persisted_notification.reference == reference
     assert not persisted_notification.personalisation
 
 
-def test_should_have_sent_status_if_fake_callback_function_fails(sample_notification, mocker):
+def test_should_not_have_sent_status_if_fake_callback_function_fails(sample_notification, mocker):
     mocker.patch("app.delivery.send_to_providers.send_sms_response", side_effect=HTTPError)
 
     sample_notification.key_type = KEY_TYPE_TEST
 
     with pytest.raises(HTTPError):
         send_to_providers.send_sms_to_provider(sample_notification)
-    assert sample_notification.status == "sent"
-    assert sample_notification.sent_by == "sns"
+    assert sample_notification.status == "created"
```

Comment:
Why is the status not the same here?

Reply:
We switched the operation order and we try to send first and then save into the database. This test provokes an error with the mocked send so it  does not get to the saving part with the notification update. This should follow more closely how the code works with the real providers (i.e. when in non research mode).